### DISCUSSION TITLE
Reverse X/Y axis in PyVista

### DISCUSSION THREAD

**Discussion by ycli0536** [pyvista/pyvista#1415] [Q&A] [answered]
Hi! I want to know if PyVista can achieve **reversing X/Y-axis dir**? I didn't find related info (words such as _'dir', 'orientation', 'mirror', 'axes'_ ...) in the document.
Currently, the X-axis starts at 0 and goes up to the max value. I would like the X-axis to start at the max value and go up to 0 (a **left-handed coordinate**).
When I add points or lines in the plotter, I also would like them to be showing in a left-handed coordinate.

THANK YOU

**Comment by akaszynski** [comment, marked as answer]
It's not clear that either `pyvista` or VTK can reverse an axis.  What you may try doing is mirroring the points of your mesh if you'd like to reverse the direction.  For example:

```py
mesh.points[:, 0] *= -1
```
It's not perfect, but the x-axis will effectively be mirrored.

**Reply by ycli0536** [reply]
Yes. It could be a solution and very easy! By transposing/rotating my value/data/attribute matrix before `flatten()`, then set `grid.cell_arrays["values"]`, I also achieve it in a not-perfect way. What all I did is reverse the X and Y-axes. Just take X-axis as Y-axis and Y-as X-axis (sth shown as below 2 figs). Now I think the main reason that I have this requirement is that I didn't declare what coordinate system should be used at first :(
![1](https://user-images.githubusercontent.com/43446843/123935738-150fac00-d9c7-11eb-98d2-570c9235c20b.png)
![2](https://user-images.githubusercontent.com/43446843/123935747-17720600-d9c7-11eb-9221-fe22085538de.png)

THANK YOU!
